libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -107,6 +107,73 @@ string ABIFunctions::tupleEncoder(
 	});
 }
 
+string ABIFunctions::tupleEncoderPacked(
+	TypePointers const& _givenTypes,
+	TypePointers const& _targetTypes
+)
+{
+	EncodingOptions options;
+	options.encodeAsLibraryTypes = false;
+	options.encodeFunctionFromStack = true;
+	options.padded = false;
+	options.dynamicInplace = true;
+
+	string functionName = string("abi_encode_tuple_");
```

Comment:
Will this name potentially collide with the non-packed version? Even if not, maybe still better to be safe:
```suggestion
	string functionName = string("abi_encode_tuple_packed_");
```

Reply:
`options.toFunctionNameSuffix` should take care of this, but perhaps it does not hurt to add it in addition to that.